This script file is used to analysis Auto.csv dataset.

In [38]:
import pandas as pd
import numpy as np

In [39]:
file = pd.read_csv('data/Auto.csv')
file.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 392 entries, 0 to 391
Data columns (total 10 columns):
index           392 non-null int64
mpg             392 non-null float64
cylinders       392 non-null int64
displacement    392 non-null float64
horsepower      392 non-null int64
weight          392 non-null int64
acceleration    392 non-null float64
year            392 non-null int64
origin          392 non-null int64
name            392 non-null object
dtypes: float64(3), int64(6), object(1)
memory usage: 30.7+ KB


In [40]:
description = file.describe()
print(description)

            index         mpg   cylinders  displacement  horsepower  \
count  392.000000  392.000000  392.000000    392.000000  392.000000   
mean   196.500000   23.445918    5.471939    194.411990  104.469388   
std    113.304898    7.805007    1.705783    104.644004   38.491160   
min      1.000000    9.000000    3.000000     68.000000   46.000000   
25%     98.750000   17.000000    4.000000    105.000000   75.000000   
50%    196.500000   22.750000    4.000000    151.000000   93.500000   
75%    294.250000   29.000000    8.000000    275.750000  126.000000   
max    392.000000   46.600000    8.000000    455.000000  230.000000   

            weight  acceleration        year      origin  
count   392.000000    392.000000  392.000000  392.000000  
mean   2977.584184     15.541327   75.979592    1.576531  
std     849.402560      2.758864    3.683737    0.805518  
min    1613.000000      8.000000   70.000000    1.000000  
25%    2225.250000     13.775000   73.000000    1.000000  
50%   

(a) Which of the predictors are quantitative, and which are qualitative?
    quantitative: "mpg", "cylinders", "displacement", "horsepower", "weight", "acceleration"
    qualitative:  "year", "origin", "name"

(b) What is the range of each quantitative predictor? You can answer this using the range() function.
    Since for every attribute the value is non-null and the min of every attribute is larger than 0, that means there is no missing data.
    
    And I can get the range from the min and max value for every attribute.
    "mpg":          9 ~ 46.6
    "cylinders":    3 ~ 8
    "displacement": 68 ~ 455
    "horsepower":   46 ~ 230
    "weight":       1613 ~ 5140
    "acceleration": 8 ~ 24.8

(c) What is the mean and standard deviation of each quantitative predictor?
                    mean          std
    "mpg":          23.445918     7.805007
    "cylinders":    5.471939      1.705783
    "displacement": 194.411990    104.644004
    "horsepower":   104.469388    38.491160
    "weight":       2977.584184   849.402560
    "acceleration": 15.541327     2.758864

(d) Now remove the 10th through 85th observations. What is the range, mean, and standard deviation of each predictor in the subset of the data that remains?